File: PopView弹出练习/Pop3-OC/Pop3-OC/ViewController.m
//
//  ViewController.m
//  Pop3-OC
//
//  Created by 王玲峰 on 4/12/20.
//  Copyright © 2020 王玲峰. All rights reserved.
//

#import "ViewController.h"
#import "POPViewController.h"

@interface ViewController ()
<UIPopoverPresentationControllerDelegate>
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}
- (IBAction)action:(UISwitch *)sender {
    POPViewController *popVC = [[POPViewController alloc]init];
    popVC.modalPresentationStyle = UIModalPresentationPopover;
//    UIModalPresentationPageSheet
//    UIModalPresentationStyle
    popVC.preferredContentSize = CGSizeMake(200, 200);
    popVC.popoverPresentationController.delegate = self;
    popVC.popoverPresentationController.sourceView = sender;
    popVC.popoverPresentationController.sourceRect = sender.bounds;
    popVC.popoverPresentationController.permittedArrowDirections = UIPopoverArrowDirectionAny;
    [self presentViewController:popVC animated:YES completion:nil];
    
}

#pragma mark - popoverdelegate
// 设置点击蒙版是否消失
- (BOOL) popoverPresentationControllerShouldDismissPopover:(UIPopoverPresentationController *)popoverPresentationController{
    return YES;
}

// 默认返回的是覆盖整个屏幕
- (UIModalPresentationStyle)adaptivePresentationStyleForPresentationController:(UIPresentationController *)controller{
    //这个 关键
    return UIModalPresentationNone;
}

// 弹出视图消失后调用的方法
- (void)popoverPresentationControllerDidDismissPopover:(UIPopoverPresentationController *)popoverPresentationController{
    NSLog(@"dismissed");
}


@end

File: PopView弹出练习/popView2/popView2/ViewController.swift
//
//  ViewController.swift
//  popView2
//
//  Created by 王玲峰 on 4/12/20.
//  Copyright © 2020 王玲峰. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
    }

    @IBAction func open(_ sender: UISwitch) {
        //1.创建控制器
        let controller = UIViewController()
        controller.view.backgroundColor = UIColor.red
        
        //2.设置b模态的效果
        controller.modalPresentationStyle = .popover
        
        //2.1 方向
        controller.popoverPresentationController?.sourceView = sender
        controller.popoverPresentationController?.sourceRect = sender.bounds
        
        //2.2 设置穿透视图
        controller.popoverPresentationController?.passthroughViews = [sender]
        
        //2.3 设置背景
        controller.popoverPresentationController?.backgroundColor = UIColor.cyan
        
        controller.preferredContentSize = CGSize(width: 300, height: 300)
        
        controller.popoverPresentationController?.delegate = (self as UIPopoverPresentationControllerDelegate)
        
        //3.弹出
        present(controller, animated: true, completion: nil)
        
    }
    
}

extension ViewController:UIPopoverPresentationControllerDelegate{
    // 默认返回的是覆盖整个屏幕
//    - (UIModalPresentationStyle)adaptivePresentationStyleForPresentationController:(UIPresentationController *)controller{
//        return UIModalPresentationNone;
//    }
    
    func adaptivePresentationStyle(for controller: UIPresentationController) -> UIModalPresentationStyle {
        return .none
    }
    
    

}

